feathr-impl/src/main/scala/com/linkedin/feathr/offline/swa/SlidingWindowAggregationJoiner.scala

```diff
@@ -164,9 +171,16 @@ private[offline] class SlidingWindowAggregationJoiner(
               failOnMissingPartition)
 
         // If skip missing features flag is set and there is a data related error, an empty dataframe will be returned.
-        if (originalSourceDf.isEmpty) {
+        if (originalSourceDf.isEmpty && shouldSkipFeature) {
           res.map(notJoinedFeatures.add)
           anchors.map(anchor => (anchor, originalSourceDf))
+        } else if (originalSourceDf.isEmpty && shouldAddDefaultColForMissingData) { // If add default col for missing data flag features
+          // flag is set and there is a data related error, an empty dataframe will be returned.
+          res.map(emptyFeatures.add)
+          val exceptionMsg = s"Missing data for features ${emptyFeatures}. Default values will be populated for this column."
```

Comment:
The notification message itself would be carved in FFE. I think feathr-engine should just send back a Map of features patched with values substituted (either default or null when default is not defined).

Reply:
Done.